core/lib/Cardano/Address/Style/Byron.hs

```diff
@@ -289,16 +293,20 @@ deriveAddressPrivateKey acctK =
 -- > "DdzFFzCqrhsq3KjLtT51mESbZ4RepiHPzLqEhamexVFTJpGbCXmh7qSxnHvaL88QmtVTD1E1sjx8Z1ZNDhYmcBV38ZjDST9kYVxSkhcw"
 
 -- | Analyze an 'Address' to know whether it's a Byron address or not.
--- Returns 'Nothing' if the address isn't a byron address, or return a
+-- Throws 'ByronAddrError' if the address isn't a byron address, or return a
 -- structured JSON that gives information about an address.
 --
 -- @since 2.0.0
-inspectByronAddress :: Address -> Maybe Json.Value
+inspectByronAddress :: MonadThrow m => Address -> m Json.Value
 inspectByronAddress addr = do
-    payload <- CBOR.deserialiseCbor CBOR.decodeAddressPayload bytes
-    (root, attrs) <- CBOR.deserialiseCbor decodePayload payload
-    path  <- find ((== 1) . fst) attrs
-    ntwrk <- CBOR.deserialiseCbor CBOR.decodeProtocolMagicAttr payload
+    payload <- either (throwM . BrDeserialiseError) pure
+      $ CBOR.deserialiseCbor CBOR.decodeAddressPayload bytes
+    (root, attrs) <- either (throwM . BrDeserialiseError) pure
+      $ CBOR.deserialiseCbor decodePayload payload
+    path  <- maybe (throwM BrInvalidDerivationPath) pure
+      $ find ((== 1) . fst) attrs
```

Comment:
Small note: the issue here isn't that the path is invalid, but that the path is missing / not present in the address attributes. I'd suggest something along the lines of `MissingExpectedDerivationPath`